pilot/pkg/networking/util/util.go

```diff
@@ -707,7 +707,7 @@ func BuildStatefulSessionFilter(svc *model.Service) *hcm.HttpFilter {
 }
 
 func MaybeBuildStatefulSessionFilterConfig(svc *model.Service) *statefulsession.StatefulSession {
-	if svc == nil {
+	if svc == nil || !features.EnablePersistentSessionFilter {
```

Comment:
We only add the main Empty session filter if this flag is enabled. If that is not added, even if you add route level it is not honoured and it is added if service has labels. So this check is not needed. Is there a problem you are trying to solve?